api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -233,4 +238,30 @@ public PatientQueue pickPatientQueue(PatientQueue patientQueue, Provider provide
         patientQueue.setQueueRoom(queueRoom);
         return dao.savePatientQueue(patientQueue);
     }
+
+    /**
+     * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#
+     * getPatientQueueByParentLocation(org.openmrs.Location,
+     * org.openmrs.module.patientqueueing.model.PatientQueue.Status, java.util.Date dateFrom,
+     * java.util.Date)
+     */
+    @Override
+    public List<PatientQueue> getPatientQueueByParentLocation(Location parentLocation, PatientQueue.Status status, Date fromDate, Date toDate, boolean onlyInQueueRooms) {
+        LocationTag queueRomTag = Context.getLocationService().getLocationTagByUuid(ROOM_TAG_UUID);
+        Set<Location> childLocations = new HashSet<>();
+        if (onlyInQueueRooms && parentLocation.getChildLocations(false) != null) {
```

Comment:
For the getChildLocations which takes a boolean, you do not have to test for NULL because it can never be so.